packages/misc/src/lib/misc.ts

```diff
@@ -859,3 +859,41 @@ export async function executeWithRetry<T>(
     requestId,
   };
 }
+
+/**
+ * Attempts to normalize a string by unescaping it until it can be parsed as a JSON object,
+ * then stringifies it exactly once. If the input is a regular string that does not represent
+ * a JSON object or array, the function will return it as is without modification.
+ * This function is designed to handle cases where strings might be excessively escaped due
```

Comment:
When do we encounter this situation?

Reply:
in the signed siwe message

Reply:
So the node returns the message containing nested "\"? How do we handle it currently? Shouldn't deserializing into Json take care of it automatically?

Reply:
@MaximusHaximus what do you think? ^

Reply:
Yea, the problem is we do not know how many times the json has been serialised, there were cases/bugs before where we would receive a json that has been serialised twice, and we spent ages to figure out why wasn't it working because the value looked correct, ended up we had to deserialise it twice.

Reply:
It feels very odd for us to need to be handling specific characters with bespoke string replacement calls in data that should be JSON (or any other standard encoding) encoded instead of just _using the decoder_ to decode it recursively.

If the problem is double-json-serialized data, could we instead check to see if the result of a `json.parse()` call returns a `string` where it should be an `object` (or `array`, or whatever we expected), then just run `parse()` on the string again to get what we need?

Reply:
Actually, I'm gonna check if this is still the case. Ideally, we should always receive a JSON string that has been serialised only once, and we shouldn't do any magic here.